### PR TITLE
bt-22: Add FormControl and FormLabel components

### LINKED ISSUE #22: Create FormControl and FormLabel components
https://mui.com/material-ui/react-text-field/#components

### PR DESCRIPTION
**Pull request by Monyster**
Closes #22 

Examples of usage:
```js
<FormControl variant="standard">
    <InputLabel htmlFor="component-simple">Name</InputLabel>
    <Input
        id="component-simple"
        defaultValue="Composed TextField"
    />
</FormControl>
```
```js
<FormControl variant="standard">
<InputLabel htmlFor="component-helper">Name</InputLabel>
<Input
  id="component-helper"
  defaultValue="Composed TextField"
  aria-describedby="component-helper-text"
/>
<FormHelperText id="component-helper-text">
  Some important helper text
</FormHelperText>
</FormControl>
```
```js
<FormControl isDisabled variant="standard">
<InputLabel htmlFor="component-disabled">Name</InputLabel>
<Input
  id="component-disabled"
  defaultValue="Composed TextField"
/>
<FormHelperText>Disabled</FormHelperText>
</FormControl>
```
```js
<FormControl isError variant="standard">
<InputLabel htmlFor="component-error">Name</InputLabel>
<Input
  id="component-error"
  defaultValue="Composed TextField"
  aria-describedby="component-error-text"
/>
<FormHelperText id="component-error-text">
  Error
</FormHelperText>
</FormControl>
```
```js
<FormControl variant="outlined">
<InputLabel htmlFor="component-outlined">Name</InputLabel>
<OutlinedInput
  id="component-outlined"
  defaultValue="Composed TextField"
  label="Name"
/>
```
```js
</FormControl>
<FormControl variant="filled">
<InputLabel htmlFor="component-filled">Name</InputLabel>
<FilledInput
  id="component-filled"
  defaultValue="Composed TextField"
/>
</FormControl>
```
```js
<FormControl
sx={{
width: '300px',
display: 'flex',
gap: '10px',
padding: '20px',
}}
>
<FormLabel>Form Label</FormLabel>

<FormControl variant="outlined">
<InputLabel htmlFor="component-outlined">Name</InputLabel>
<OutlinedInput id="component-outlined" label="Name" />
</FormControl>

<FormControl variant="filled">
<InputLabel htmlFor="component-filled">Email</InputLabel>
<FilledInput id="component-filled" />
</FormControl>
</FormControl>
```

Example image:
image

### CODE REVIEW

File: frontend/src/bundles/common/components/form-control/form-control.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    className?: string;
+    sx?: SxProps<Theme>;
+};
+
+const FormControl: React.FC<Properties> = ({
+    children,
+    color = 'primary',
+
+    variant = 'standard',
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common/components/form-control/form-control.tsx]: In this case you should use your object values:

```suggestion

    variant = InputVariantTypes.STANDART,
```

File: frontend/src/bundles/common/components/form-control/form-control.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import { FormControl as MUIFormControl } from '@mui/material';
+
+const InputVariantTypes = {
+    FILLED: 'filled',
+    OUTLINED: 'outlined',
+    STANDART: 'standard',
+} as const;
+
+type FormControlVariants =
+    (typeof InputVariantTypes)[keyof typeof InputVariantTypes];
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common/components/form-control/form-control.tsx]: In this case you need to use ValueOf from '~/bundles/common/types/types.js';

File: frontend/src/bundles/common/components/form-control/form-control.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
+import { type SxProps, type Theme } from '@mui/material';
+import { FormControl as MUIFormControl } from '@mui/material';
+
+const InputVariantTypes = {
+    FILLED: 'filled',
+    OUTLINED: 'outlined',
+    STANDART: 'standard',
+} as const;
```
Comment (MaksymGavatuk) [inline frontend/src/bundles/common/components/form-control/form-control.tsx]: Move it to enums folder

**Review by MaksymGavatuk — CHANGES_REQUESTED** [review]

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
export components in the root components.ts file

File: frontend/src/bundles/common/components/form-control/form-control.tsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+type Properties = MUIFormControlProperties & {
+    isError?: boolean;
+    isDisabled?: boolean;
+};
+
+const FormControl: React.FC<Properties> = ({
+    children,
+    color = ColorProperty.PRIMARY,
+    variant = InputVariant.STANDARD,
+    isError = false,
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/form-control/form-control.tsx]:
```suggestion
    hasError = false,
```

**Review by nikita-remeslov — APPROVED** [review]

File: frontend/src/bundles/common/components/form-label/form-label.tsx [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { type FormLabelProps as MUIFormLabelProperties } from '@mui/material';
+import { FormLabel as MUIFormLabel } from '@mui/material';
+
+type Properties = MUIFormLabelProperties;
+
+const FormLabel: React.FC<Properties> = ({
+    children,
+    className = '',
+    ...properties
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/form-label/form-label.tsx]:
```suggestion
    ...props
```

File: frontend/src/bundles/common/components/form-label/form-label.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+}) => (
+    <MUIFormLabel className={className} {...properties}>
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/form-label/form-label.tsx]:
```suggestion
    <MUIFormLabel className={className} {...props}>
```

**Review by JosuerBague — COMMENTED** [review]
Looks good to me.

**Review by MaksymGavatuk — APPROVED** [review]

**Comment by nikita-remeslov** [comment]
Fix conflicts and we can merge

File: frontend/src/bundles/common/components/components.ts [hunk 7]
```diff
@@ -1,6 +1,7 @@
 export { App } from '../../../app/app.js';
 export { Button } from './button/button.js';
 export { Checkbox } from './checkbox/checkbox.js';
+export { FormControl } from './form-control/form-control.js';
```
Comment (ptarho) [inline frontend/src/bundles/common/components/components.ts:5]: You forget to export FormLabel

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**